app/helpers/authentication_helper.rb

```diff
@@ -15,8 +15,11 @@ def authentication_enabled_providers
     end
   end
 
-  def current_user_authentication_enabled_providers
-    providers = Authentication::Providers.enabled & current_user.identities.pluck(:provider).map(&:to_sym)
+  # Returns the list of enabled providers for the given user, defaults to `current_user`
+  def authentication_enabled_providers_for_user(user = nil)
+    user ||= current_user
+
+    providers = Authentication::Providers.enabled & user.identities.pluck(:provider).map(&:to_sym)
```

Comment:
I'm starting to wonder if this is really a helper, it doesn't seem to be directly related to view concerns. How about moving this into `Authentication::Providers.enabled_for_user(user)`? You could call that directly in `app/views/layouts/mailer.html.erb`, `app/views/users/_account.html.erb` and `app/views/users/_account_providers_settings.html.erb`, or wrap it up in helper that doesn't do more than pass through `current_user`, e.g.

```ruby
def authentication_enabled_providers_for_user(user = current_user)
  Authentication::Providers.enabled_for_user(user)
end
```

Reply:
done!

Reply:
Thanks @rhymes!